### ISSUE TITLE
error TS1239: Unable to resolve signature of parameter decorator when called as an expression

### ISSUE THREAD

**Issue by khteh** [remojansen/InversifyJS#1505]
```
import { EntityTarget, Repository } from "typeorm"
import { injectable, unmanaged, inject } from "inversify";

@injectable()
export abstract class RepositoryBase<T extends EntityBase> implements IRepository<T> {
    protected _repository: Repository<T>;
    protected _db: Database;
    constructor(@unmanaged() entity: EntityTarget<T>, db: Database) {
        this._db = db;
        this._repository = this._db.getRepository(entity);
    }
```
## Expected Behavior
No build error
## Current Behavior
```
Data/Repositories/RepositoryBase.ts:11:18 - error TS1239: Unable to resolve signature of parameter decorator when called as an expression.
  Argument of type 'undefined' is not assignable to parameter of type 'string'.

11     constructor(@unmanaged() entity: EntityTarget<T>, db: Database) {
                    ~~~~~~~~~~~


Found 1 error in Data/Repositories/RepositoryBase.ts:11
```
## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used:
* Environment name and version (e.g. Chrome 39, node.js 5.4): v19.8.1
* Operating System and version (desktop or mobile): Ubuntu 22.10
* Link to your project:

# Stack trace

**Comment by thatsmydoing** [comment]
It seems this PR would address the issue https://github.com/inversify/InversifyJS/pull/1499

**Comment by josendf-basetis** [comment]
Hi,

Any update on this issue? Blocks upgrading to Typescript v5.

Thank you for your help,

**Comment by PodaruDragos** [comment]
@josendf-basetis  this is the update [link](https://github.com/inversify/InversifyJS/pull/1499#issuecomment-1625217581).

**Comment by josendf-basetis** [comment]
Thank you for your help @PodaruDragos .

Now that the fix has been merged, is there any prevision  on npm package release date?

**Comment by Jameskmonger** [comment]
Closing as fixed, this was published in [release `6.0.2`](https://github.com/inversify/InversifyJS/commit/5d3d62afc30a956a0589d4871bf98f07002d53f9) on 2023/10/21.

Please don't hesitate to reopen this issue if you find it persists.

cc @khteh @thatsmydoing @josendf-basetis @dennisameling @samir-ayoub @vlzuiev

cc everyone who reacted to original issue: @janeisenmenger @danewilson @mscharley @RGBWang @restfulhead @zhouguoqing917 @dribba
